ludwig/config_validation/checks.py

```diff
@@ -344,10 +348,14 @@ def check_tagger_decoder_requirements(config: "ModelConfig") -> None:  # noqa: F
 @register_config_check
 def check_hyperopt_parameter_dicts(config: "ModelConfig") -> None:  # noqa: F821
     """Checks for hyperopt parameter dicts against their config objects."""
-    from ludwig.schema.hyperopt.utils import get_parameter_cls, parameter_config_registry
+    if config.hyperopt is None:
+        return
+
+    from ludwig.schema.hyperopt.utils import get_parameter_cls, parameter_config_registry  # noqa: F401
 
-    if config.hyperopt is not None:
-        for parameter, space in config.hyperopt.parameters.items():
+    for parameter, space in config.hyperopt.parameters.items():
+        # skip nested hyperopt parameters
+        if parameter != ".":
```

Comment:
Confused by this. When would parameter be a single period?

Reply:
When it's a nested space. According to the docs, we name nested spaces `.` in the config.

Reply:
@ksbrar This was functionality that Travis added [a while ago](https://ludwig.ai/latest/user_guide/hyperopt/#nested-ludwig-config-parameters).

Reply:
Ah gotcha! Thanks for the link.